src/tools/rbd/action/Bench.cc

```diff
@@ -279,10 +298,10 @@ int do_bench(librbd::Image& image, io_type_t io_type,
       start_pos = (rand() % (size / io_size)) * io_size;
       break;
     case IO_PATTERN_SEQ:
-      start_pos = seq_chunk_length * i;
+      start_pos = (io_start + seq_chunk_length * i) % size;
       break;
     case IO_PATTERN_FULL_SEQ:
-      start_pos = i * io_size;
+      start_pos = io_start + i * io_size;
```

Comment:
... and here no attempt to wrap based on image size is made at all?

Reply:
Agree. Code was already missing a check that i * io_size was within the image size so I'll fix that as well.